main.go

```diff
@@ -75,13 +76,16 @@ func main() {
 					index = fmt.Sprintf("%s-%s", "aleph", ft)
 				}
 
-				// if a file path is passed as a flag
-				if c.Args().Get(0) != "-" {
-					// Open the file.
-					file, err = os.Open(c.Args().Get(0))
+				inputData := c.Args().Get(0)
+				if len(inputData) == 0 {
```

Comment:
There are a few things here I think we should change. We should probably support s3 object names with slashes (ex: `s3://foo/bar/baz.mrc`) and we should probably support local file names that start with `s3` (ex: `s3-download.mrc`). I would suggest replacing this logic with something like:

```
if strings.HasPrefix(inputData, "s3://") {
    s3Info := strings.SplitAfterN(inputData[5:], "/", 2)
    file, err = getS3Obj(s3Info[0], s3Info[1])
} else {
...
```

Reply:
Tracking under: https://mitlibraries.atlassian.net/browse/DIP-233